### ISSUE TITLE
Batch updates for pull requests

### ISSUE THREAD

**Issue by jantman** [isaacs/github#340]
If a large PR is submitted and someone comments on 20 different lines, everyone watching that repo gets 20 emails about comments. There should be a way to batch these up - i.e. a button that allows me to start a "review session", make as many comments as I want, and then "submit" my session - sending all those comments as one batch.

**Comment by cirosantilli** [comment]
I would add the word "notification" to the title instead of update.

**Comment by nevir** [comment]
This is more than just wanting batched notifications, I think. Having a set of comments come in as a batch is a handy workflow:

* The reviewer can build up a set of comments (drafts).
* Once they are happy with their comments, they publish them; email is sent out at this point, and the comments are made visible.
* It also serves as an explicit "ok, I'm done reviewing for now"

**Comment by lotyrin** [comment]
Definitely would like batching on send side for reviews, but clustering+digest on receive side is critical.

**Comment by TroyAlchemy** [comment]
Even if it just waited 10 mins to allow more comments to be added and then emailed, it would be better.

**Comment by salsanchez** [comment]
👍

**Comment by benmeuer** [comment]
:+1:

**Comment by mattodd** [comment]
Yes, would love to have this for Open Source Malaria, and I'm sure lots of other groups would welcome it.

**Comment by chamini2** [comment]
This is resolved with reviews, right?

**Comment by hopeseekr** [comment]
I received 6,532 emails from GitHub.com from 17 Sep 2017 and 5 Feb 2018 (141 days).

That's **forty-six (46.33) emails a day*! 

Except, very few came in during the weekends. According to [The Business Days Calculator](https://www.timeanddate.com/date/workdays.html?d1=17&m1=9&y1=2017&d2=6&m2=2&y2=2018), that's 95 days.

For the last 5 months, I've been receiving **close to 69 (68.76) GitHub emails every day!**

It's intolerable. Almost all of the emails are from notifications of commits to my ~15 co-workers' forks of the main repository.

**Comment by JohannesMP** [comment]
Please add an option on a repo's Watch settings to send daily/weekly digest emails for notifications. This should include Issues, pull requests, etc.

**Comment by KopfKrieg** [comment]
Any update on this? I need to get notifications for a lot of repos but I don't have to get a separate e-mail for every single one. Having a digest/summary setting for notifications woudl be really cool. And it would also reduce stress a lot here.

**Comment by marc-hb** [comment]
> This is resolved with reviews, right?

Very partially. github reviews are just the start of the review conversations, then it's back to one email per comment unless I missed something.
If every comment in every situation could always be made part of a larger github review then yes, that would help a lot.

**Comment by marc-hb** [comment]
Now I wonder whether the impossibility to batch reviews again isn't caused by the use of force-pushes...

As opposed to force pushes, "fixup" commits and final squash seems to be the model github focuses on. I wish the documentation made that point louder and clearer because it's not how most other git reviews work.
